services/py-api/tests/integration_tests/test_jwt_utility.py

```diff
@@ -0,0 +1,82 @@
+import os
+from unittest.mock import patch
+from result import Err, Ok
+from src.server.schemas.jwt_schemas.jwt_user_data_schema import JwtUserData
+from datetime import datetime, timedelta, timezone
+from src.utils import JwtUtility
+
+sufficient_expiration_time = (datetime.now(tz=timezone.utc) + timedelta(seconds=30)).timestamp()
+instant_expiration_time = (datetime.now(tz=timezone.utc)).timestamp()
+
+CORRECT_TEST_PAYLOAD = JwtUserData(
+    sub="alksdjflksd987fsidjf98sduf", is_admin=True, team_id="slkdjflkasjdflkjsdlkj", exp=sufficient_expiration_time
+)
+
+MISSING_KEY_TEST_PAYLOAD = JwtUserData(  # We need it only for testing purposes | mypy complains
+    sub="alksdjflksd987fsidjf98sduf", is_admin=True, team_id="slkdjflkasjdflkjsdlkj"  # type: ignore[typeddict-item]
+)
+
+ADDITIONAL_KEY_TEST_PAYLOAD = JwtUserData(
+    sub="alksdjflksd987fsidjf98sduf", is_admin=True, team_id="slkdjflkasjdflkjsdlkj", exp=sufficient_expiration_time, name="abcdefgji"  # type: ignore[typeddict-unknown-key]
+)
+
+EXPIRED_TEST_PAYLOAD = JwtUserData(
+    sub="alksdjflksd987fsidjf98sduf", is_admin=True, team_id="slkdjflkasjdflkjsdlkj", exp=instant_expiration_time
+)
+
+
+@patch.dict("os.environ", {"SECRET_KEY": "abcdefghijklmnopqrst"})
```

Comment:
Is this patched correctly and why do we need to patch it? Also, as this is repeated maybe extract it as a variable.

Reply:
This type of patching works because if you see the test passes when you try encoding/decoding with different keys. I do this so that we dont have to include the SECRET_KEY as a secret in github workflows so that the tests pass within the github actions virtual machines where are repository is mounted and tests are run

Reply:
Correct